### PR TITLE
future-proof HTTP

### LINKED ISSUE #334: http package should not go back to drand node for rounds in the future
that could be a dos vector if it does

### PR DESCRIPTION
**Pull request by willscott**
fix #334

### CODE REVIEW

File: http/server.go [hunk 1]
```diff
@@ -182,6 +182,15 @@ func (h *handler) PublicRand(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	h.pendingLk.RLock()
```
Comment (nikkolasg) [inline http/server.go:185]: I think it's totally fine with a rlock but I was just thinking that it is not necessary if you only look at the current time, so there's no need to have the http server "stateful" (and no lock).

Comment (willscott) [reply http/server.go:185]: we need to look at the state to know if we have the state :)

Comment (nikkolasg) [reply http/server.go:185]: ? No I'm saying that you could only look at the current time `time.Now()` and look if the request contains a round higher than the round at the current time. No state here.

Comment (nikkolasg) [reply http/server.go:185]: But note I have approved that PR :)

**Comment by willscott** [comment]
went to change this to be cleaner, but it looks like i got a bit too close to the kyber math :)

@nikkolasg - if you get a chance, any idea why changing the genesis in the mock grpc server such that the returned round is seen as being the one that should happen 'now' causes the signatures to fail to verify?

**Comment by nikkolasg** [comment]
That's because of this [line](https://github.com/drand/drand/blob/master/test/mock/grpcserver.go#L63): for a special round, the mock server returns an invalid signature, and by moving the genesis time, it seems you fall on this one (i've checked and the first request falls on round 1970 which is exactly triggering the special case). 
Feel free to remove the special case if you want.
If you need a server that returns a new signature over the round, I can give you that as well.

**Comment by willscott** [comment]
> That's because of this [line](https://github.com/drand/drand/blob/master/test/mock/grpcserver.go#L63): for a special round, the mock server returns an invalid signature, and by moving the genesis time, it seems you fall on this one (i've checked and the first request falls on round 1970 which is exactly triggering the special case).
> Feel free to remove the special case if you want.
> If you need a server that returns a new signature over the round, I can give you that as well.

i think there's something more going on. That line triggers one of the signature verification rounds to fail with "invalid length". However, the round before and after are failing with an error of "bls: invalid signature", which is the part i wasn't expecting.

**Comment by nikkolasg** [comment]
Which test is failing that shows the invalid bls ? I only looked at `client/TestHTTPWatch`

**Comment by nikkolasg** [comment]
I only see the problem with the length of the signature:
```bash
=== RUN   TestHTTPWatch
ts="Tue, 19 May 2020 16:19:01 BST" call=http.go:156 level=warn http_client="failed to
 verify value" err="input string should be equal or larger than 96"
ts="Tue, 19 May 2020 16:19:01 BST" call=http_test.go:109 level=error polling_client="
failed to watch" err="input string should be equal or larger than 96"
    TestHTTPWatch: http_test.go:112: Should get a result from watching
--- FAIL: TestHTTPWatch (0.04s)
FAIL
exit status 1
FAIL    github.com/drand/drand/client   0.051s
```

**Pull request merged**